src/ring_theory/chain_of_divisors.lean

```diff
@@ -0,0 +1,272 @@
+/-
+Copyright (c) 2021 . All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen, Paul Lezeau
+-/
+
+import algebra.is_prime_pow
+import algebra.squarefree
+
+/-!
+
+# Chains of divisors
+
+The results in this file show that in the monoid `associates M` of a `unique_factorization_monoid`
+`M`, an element `a` is an n-th prime power iff its set of divisors is a strictly increasing chain
+of length `n + 1`, aka we can find a strictly increasing bijection bewteen `fin (n + 1)` and
+the set of factors of `a`.
+
+We can then use this to show that given `a b : M`, if there is an monotone bijection
+between the sets of factors of `associates.mk a` and `associates.mk b`, then the prime
+factorisations of `a` and `b` have the same shape.
```

Comment:
Can you mention here the results that contain these statements?

Reply:
These results aren't in this PR, so I've removed these lines for now.